extensions/jms/src/main/java/org/mule/extensions/jms/api/source/JmsListener.java

```diff
@@ -318,9 +268,25 @@ private boolean replyDestinationIsTopic(Destination destination) {
     return destination instanceof Topic;
   }
 
-  private void waitForMessageToBeProcesed() {
-    if (synchronous) {
-      jmsLock.lock();
+  private JmsListenerLock createJmsLock() {
+    return synchronous ? new DefaultJmsListenerLock() : new NullJmsListenerLock();
+  }
+
+  private int getValidNumberOfConsumers(int numberOfConsumers) {
+    if (numberOfConsumers > 1 && consumerType.isTopic()) {
+      TopicConsumer topicConsumer = (TopicConsumer) consumerType;
+
+      if (!isCapableOfMultiConsumersOnTopic(topicConsumer)) {
+        LOGGER.warn("Destination [" + destination + "] is a topic, but " + numberOfConsumers
+            + " receivers have been requested. Will configure only 1");
```

Comment:
let's also hint that this is only possible if the topic consumer is Shared:
"Destination [" + destination + "] is a topic, but " + numberOfConsumers  + " receivers have been requested. This is only possible for 'shared' topic consumers. Only one will be configured instead."